WikiExtension/src/timeSeriesService.js

```diff
@@ -25,24 +26,97 @@ const fetchPageViews = async (title, startDate, endDate, aggregateType) => {
     return await response.json();
 };
 
+/**
+ * Get all the revision objects on a given wikipedia title.
+ *
+ * The api only gives us at most 20 reivision objects per request. However, it does give us a query to
+ * use if there are any older revisions. If older versions of revision exists, we will get
+ * {
+ *  revisions: [...],
+ *  ...
+ *  older: 'https://en.wikipedia.org/w/rest.php/v1/page/Jupiter/history?older_than=1103899458',
+ *  ...
+ * }
+ *
+ * Thus, we keep calling the endpoint with the given query if the older property exists in the
+ * previous request.
+ *
+ * @param {string} title of a wikipedia article
+ * @param {Date} startDate
+ * @param {Date} endDate
+ * @returns An array containing all revision objects
+ */
+const fetchPageRevisions = async (title, startDate, endDate) => {
+    const response = await fetch(`https://en.wikipedia.org/w/rest.php/v1/page/${title}/history`);
+    let curJsonResponse = await response.json();
+
+    let revisions = curJsonResponse.revisions;
+
+    while (curJsonResponse.hasOwnProperty("older")) {
+        const localResponse = await fetch(curJsonResponse.older);
+        curJsonResponse = await localResponse.json();
+
+        const localRevisions = curJsonResponse.revisions;
+
+        // Stop making requests when requesting passes startDate
+        const localLatestTimestamp = new Date(localRevisions[0].timestamp);
+        if (localLatestTimestamp < startDate) break;
+
+        revisions.push(...localRevisions);
+    }
+
+    // Filters revisions that are in between the given timerange
```

Comment:
Why do we need this? You are already stopping making the request when in passes start date, and we are keeping the other range in the present

Reply:
The stopping only uses a timestamp from one of the 20 revisions (the latest of the 20 that we requested) to check if it passes the start date. It is possible that the previous revisions also include something that already passes the start date.  For the end date, we wouldn't really need it if we are fixing the end date to present, but I thought it would be helpful to just have the end date as a parameter.

Reply:
Ok yes! Thank you

Reply:
https://github.com/sukritsangvong/wp-browser-extension/blob/sangvongp/revision-count-endpoint/WikiExtension/src/timeSeriesService.js#L61-L63